// @genkit-ai/vertexai/lib/vectorsearch/vector_search/query_public_endpoint.mjs
import { logger } from "genkit/logging";
async function queryPublicEndpoint(params) {
  const {
    featureVector,
    neighborCount,
    accessToken,
    indexEndpointId,
    publicDomainName,
    projectNumber,
    deployedIndexId,
    location
  } = params;
  const url = new URL(
    `https://${publicDomainName}/v1/projects/${projectNumber}/locations/${location}/indexEndpoints/${indexEndpointId}:findNeighbors`
  );
  const requestBody = {
    deployed_index_id: deployedIndexId,
    queries: [
      {
        datapoint: {
          datapoint_id: "0",
          feature_vector: featureVector
        },
        neighbor_count: neighborCount
      }
    ]
  };
  const response = await fetch(url, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${accessToken}`
    },
    body: JSON.stringify(requestBody)
  });
  if (!response.ok) {
    logger.error("Error querying index: ", response.statusText);
  }
  return response.json();
}
export {
  queryPublicEndpoint
};
